src/main/java/com/myresearchs/swaggerinspring/swaggerinspringexample/controllers/ClientRestController.java

```diff
@@ -0,0 +1,48 @@
+package com.myresearchs.swaggerinspring.swaggerinspringexample.controllers;
+
+import com.myresearchs.swaggerinspring.swaggerinspringexample.model.Client;
+import com.myresearchs.swaggerinspring.swaggerinspringexample.services.ClientService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+//Нужно ли нам возвращать ResponseEntity?
+@Api(value = "Client Rest Controller", description = "REST APIs for Client entity.")
+@RestController
+@RequestMapping("/client")
+public class ClientRestController {
+
+    private ClientService service;
```

Comment:
А почему через конструтор?
Команда Spring главным образом выступает за инъекцию через сеттеры, потому что большое количество аргументов конструктора может стать громоздким, особенно если свойства являются необязательными. Сеттеры также делают объекты этого класса пригодными для реконфигурации или повторной инъекции позже. Управление через JMX MBeans является ярким примером

Некоторые пуристы предпочитают инъекцию на основе конструктора. Предоставление всех зависимостей означает, что объект всегда возвращается в вызывающий код в полностью инициализированном состоянии. Недостатком является то, что объект становится менее поддающимся реконфигурации и повторной инъекции

Reply:
Constructor-based or setter-based DI?

Since you can mix both, Constructor- and Setter-based DI, it is a good rule of thumb to use constructor arguments for mandatory dependencies and setters for optional dependencies.
https://docs.spring.io/spring/docs/3.1.x/spring-framework-reference/html/beans.html